src/ReferenceList.php

```diff
@@ -220,25 +177,94 @@ public function getReference( $referenceHash ) {
 	}
 
 	/**
-	 * @see Serializable::serialize
+	 * Returns the index of a reference or false if the reference could not be found.
+	 *
+	 * @since 0.5
+	 *
+	 * @param Reference $reference
+	 *
+	 * @return int|boolean
+	 */
+	public function indexOf( Reference $reference ) {
+		foreach ( $this->references as $index => $ref ) {
+			if ( $ref->equals( $reference ) ) {
+				return $index;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * @see IteratorAggregate::getIterator
 	 *
-	 * @since 2.1
+	 * @since 4.0
+	 *
+	 * @return Traversable
+	 */
+	public function getIterator() {
+		return new ArrayIterator( $this->references );
+	}
+
+	/**
+	 * @since 4.0
+	 *
+	 * @return Reference[] Numerically indexed (non-sparse) array.
+	 */
+	public function toArray() {
+		return $this->references;
+	}
+
+	/**
+	 * @see Countable::count
+	 *
+	 * @since 4.0
+	 *
+	 * @return int
+	 */
+	public function count() {
+		return count( $this->references );
+	}
+
+	/**
+	 * The hash is purely value based, ignoring the order of the elements in the array.
+	 *
+	 * @see Hashable::getHash
+	 *
+	 * @since 4.0
 	 *
 	 * @return string
 	 */
-	public function serialize() {
-		return serialize( iterator_to_array( $this ) );
+	public function getHash() {
+		$hashes = array();
+
+		foreach ( $this->references as $reference ) {
+			$hashes[] = $reference->getHash();
+		}
+
+		sort( $hashes );
+
+		return implode( '|', $hashes );
```

Comment:
This is a breaking change. The old algorithm used MapValueHasher, which used sha1. This one does not, possibly creating strings that are thousands of characters long.